### PR TITLE
feat: allow custom precision for volume grid building

### PR DESCRIPTION
**Pull request by asalzburger**
This PR allows to set some numerical precision for the volume grid binning, which removes some odd cases when writing out to Json and reading into Detray.

### CODE REVIEW

**Comment by paulgessinger** [comment]
Fails unit test at the moment.

**Comment by asalzburger** [comment]
Hey @niermann999  - this is a PR that allows to catch for numerical instabilities in the volume grid.

**Comment by asalzburger** [comment]
Ping @niermann999

File: Core/include/Acts/Detector/detail/CylindricalDetectorHelper.hpp [hunk 1]
```diff
@@ -187,6 +189,13 @@ std::array<std::vector<ActsScalar>, 3u> rzphiBoundaries(
 
   // The return boundaries
   std::array<std::set<ActsScalar>, 3u> uniqueBoundaries;
+  auto insertWithPrecision = [&](size_t is, ActsScalar value) -> void {
```
Comment (niermann999) [inline Core/include/Acts/Detector/detail/CylindricalDetectorHelper.hpp:192]:
```suggestion
  auto insertWithPrecision = [&](std::size_t is, ActsScalar value) -> void {
```

**Review by niermann999 — DISMISSED** [review]
Sorry, this had slipped my mind. What issue did it fix?

**Comment by asalzburger** [comment]
> Sorry, this had slipped my mind. What issue did it fix?

It makes rounded values for common boundaries and hence avoid that the `json` writer invents artificial boundary values with epsilon apart.

**Review by niermann999 — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [e8ff45a1130a85cdaddb92f1449b60f2aefd1b8f]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820917) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820916) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820915) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820914) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820913) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820912) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820911) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820910) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820909) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820908) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820907) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820906) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820905) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820904) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820903) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820902) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820901) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33820900) |
